src/Cli/dotnet/commands/dotnet-new/PostActions/DotnetAddJsonPropertyPostActionProcessor.cs

```diff
@@ -0,0 +1,235 @@
+
+            if (sourceNameReplaceValue != null)
+            {
+                newJsonPropertyName = newJsonPropertyName.Replace(sourceNameReplaceValue.Value.sourceName,
+                                                                  sourceNameReplaceValue.Value.replaceValue);
+
+                newJsonPropertyValue = newJsonPropertyValue.Replace(sourceNameReplaceValue.Value.sourceName,
+                                                                    sourceNameReplaceValue.Value.replaceValue);
+            }
+
+            JsonNode? newJsonContent = AddElementToJson(
+                environment.Host.FileSystem,
+                jsonFiles[0],
+                parentProperty,
+                ":",
+                newJsonPropertyName,
+                newJsonPropertyValue);
+
+            if (newJsonContent == null)
+            {
+                return false;
+            }
+
+            environment.Host.FileSystem.WriteAllText(jsonFiles[0], newJsonContent.ToJsonString(s_serializerOptions));
+
+            Reporter.Output.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Succeeded, jsonFileName));
+
+            return true;
+        }
+
+        private static JsonNode? AddElementToJson(IPhysicalFileSystem fileSystem, string targetJsonFile, string? propertyPath, string propertyPathSeparator, string newJsonPropertyName, string newJsonPropertyValue)
+        {
+            JsonNode? jsonContent = JsonNode.Parse(fileSystem.ReadAllText(targetJsonFile), nodeOptions: null, documentOptions: s_deserializerOptions);
+
+            if (jsonContent == null)
+            {
+                return null;
+            }
+
+            JsonNode? parentProperty = FindJsonNode(jsonContent, propertyPath, propertyPathSeparator);
+
+            if (parentProperty == null)
+            {
+                Reporter.Error.WriteLine(LocalizableStrings.PostAction_ModifyJson_Error_ParentPropertyPathInvalid);
+                return null;
+            }
+
+            try
+            {
+                parentProperty[newJsonPropertyName] = JsonNode.Parse(newJsonPropertyValue);
+            }
+            catch (JsonException)
+            {
+                parentProperty[newJsonPropertyName] = newJsonPropertyValue;
+            }
+
+            return jsonContent;
+        }
+
+        private static JsonNode? FindJsonNode(JsonNode content, string? nodePath, string pathSeparator)
+        {
+            if (nodePath == null)
+            {
+                return content;
+            }
+
+            string[] properties = nodePath.Split(pathSeparator);
+
+            JsonNode? node = content;
+
+            foreach (string property in properties)
+            {
+                if (node == null)
+                {
+                    return null;
+                }
+
+                node = node[property];
+            }
+
+            return node;
+        }
+
+        private static IReadOnlyList<string> FindFilesAtOrAbovePath(
```

Comment:
since this method is anyway local, is there any use of it when `maxAllowedAboveDirectories` is not 1?